src/FsAutoComplete.Core/Project.fs

```diff
@@ -23,23 +83,47 @@ type Project (projectFile, onChange: ProjectFilePath -> unit) =
                 r.Reply response
                 return! loop (lastWriteTime, response)
             | SetResponse r ->
-                return! loop (lastWriteTime, r)
-        }    
-        loop (File.GetLastWriteTimeUtc projectFile, None)
+                persistentCache.SaveCache(lastWriteTime, r)
 
-    let fullPath = Path.GetFullPath projectFile
+                return! loop (lastWriteTime, r)
+        }
+        let lwt =
+            if File.Exists projectAsset then
+                let a = File.GetLastWriteTimeUtc projectAsset
+                let p = File.GetLastWriteTimeUtc projectFile
+                if a > p then a else p
+            else
+                File.GetLastWriteTimeUtc projectFile
+        let state = persistentCache.LoadCache lwt
```

Comment:
I haven't really changed invalidation logic.It was using time stamp before (for invalidation of the `fsproj` in memory cache, I've just extended functionality to include `obj/project.assets.json`. While definitely not perfect it seemed to work well enough for old `fsproj` cache.

>  timestamp are annoying, git change these too.

We're using last write time so it should be fine from git point of view. 

> For example the fsproj can change and project.assets.json remain the same, but project info need to be invalidated. Scenario: i change a Property in fsproj (this doesnt change the project.assts.json) that will conditionall include some compiles or compiler arguments. hash check cover that, timestamp not

Yeah, and that will be handled by this logic - it invalidates whether either `fsproj` or `obj/projecgt.assets.json` change.